generator/test_case_generator.go

```diff
@@ -36,69 +36,85 @@ type testConfig struct {
-		{"./test/metric_dimension", ""},
+		{testDir: "./test/ca_bundle"},
+		{testDir: "./test/cloudwatchlogs"},
+		{testDir: "./test/metrics_number_dimension"},
+		{testDir: "./test/metric_value_benchmark"},
+		{testDir: "./test/run_as_user"},
+		{testDir: "./test/collection_interval"},
+		{testDir: "./test/metric_dimension"},
 	},
 	/*
 		You can only place 1 mac instance on a dedicate host a single time.
 		Therefore, limit down the scope for testing in Mac since EC2 can be done with Linux
 		and Mac under the hood share similar plugins with Linux
 	*/
 	"ec2_mac": {
-		{"../../../test/feature/mac", ""},
+		{testDir: "../../../test/feature/mac"},
 	},
 	"ec2_windows": {
-		{"../../../test/feature/windows", ""},
+		{testDir: "../../../test/feature/windows"},
 	},
 	"ec2_performance": {
-		{"../../test/performance/emf", ""},
-		{"../../test/performance/logs", ""},
-		{"../../test/performance/system", ""},
-		{"../../test/performance/statsd", ""},
-		{"../../test/performance/collectd", ""},
+		{testDir: "../../test/performance/emf"},
+		{testDir: "../../test/performance/logs"},
+		{testDir: "../../test/performance/system"},
+		{testDir: "../../test/performance/statsd"},
+		{testDir: "../../test/performance/collectd"},
 	},
 	"ec2_stress": {
-		{"../../test/stress/emf", ""},
-		{"../../test/stress/logs", ""},
-		{"../../test/stress/system", ""},
-		{"../../test/stress/statsd", ""},
-		{"../../test/stress/collectd", ""},
+		{testDir: "../../test/stress/emf"},
+		{testDir: "../../test/stress/logs"},
+		{testDir: "../../test/stress/system"},
+		{testDir: "../../test/stress/statsd"},
+		{testDir: "../../test/stress/collectd"},
 	},
 	"ecs_fargate": {
-		{"./test/ecs/ecs_metadata", ""},
+		{testDir: "./test/ecs/ecs_metadata"},
 	},
 	"ecs_ec2_daemon": {
-		{"./test/metric_value_benchmark", ""},
-		{"./test/statsd", ""},
-		{"./test/emf", ""},
+		{testDir: "./test/metric_value_benchmark"},
+		{testDir: "./test/statsd"},
+		{testDir: "./test/emf"},
 	},
 	"ec2_acceptance": {
-		{"./test/acceptance", ""},
+		{testDir: "./test/acceptance"},
 	},
 	"ec2_userdata": {
-		{"./test/userdata", ""},
-  },
-
+		{testDir: "./test/userdata"},
+	},
 	"eks_daemon": {
-		{"./test/metric_value_benchmark", ""},
-		{"./test/statsd", "terraform/eks/daemon/statsd"},
-		{"./test/emf", "terraform/eks/daemon/emf"},
+		{
+			testDir: "./test/metric_value_benchmark",
+			targets: map[string]map[string]struct{}{"arc": {"amd64": {}}},
+		},
+		{
+			testDir: "./test/statsd", terraformDir: "terraform/eks/daemon/statsd",
+			targets: map[string]map[string]struct{}{"arc": {"amd64": {}}},
+		},
+		{
+			testDir: "./test/emf", terraformDir: "terraform/eks/daemon/emf",
+			targets: map[string]map[string]struct{}{"arc": {"amd64": {}}},
+		},
+		{
+			testDir: "./test/fluent", terraformDir: "terraform/eks/daemon/fluent/d",
+			targets: map[string]map[string]struct{}{"arc": {"amd64": {}}},
+		},
+		{testDir: "./test/fluent", terraformDir: "terraform/eks/daemon/fluent/bit"},
```

Comment:
Any reason we run fluentbit on both arm & amd but fluentd only on amd?

Reply:
I don't know the history of the original tests, but only fluent-bit has both ARM and AMD covered.

Reply:
Hmm feels like we should be consistent for all of these infact.. but not a blocker if thats existing behavior.